core/src/main/java/org/cru/contentscoring/core/service/impl/SyncScoreServiceImpl.java

```diff
@@ -50,12 +55,34 @@ private void updateScore(
                 node.setProperty(ContentScoreUpdateServiceImpl.CONTENT_SCORE_UPDATED, now);
                 node.setProperty("cq:lastModified", now);
                 node.setProperty("cq:lastModifiedBy", "scale-of-belief");
-                Session session = resourceResolver.adaptTo(Session.class);
 
-                if (session != null) {
-                    session.save();
-                }
+                TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+                List<Tag> newTags = buildTagsWithScore(contentResource, tagManager, score);
+                tagManager.setTags(contentResource, newTags.toArray(new Tag[0]));
```

Comment:
does `setTags()` internally call `session.save()`?

Reply:
It does.